### ISSUE TITLE
anomaly "Not enough lambda/let's." on rewrite with lambda

### ISSUE THREAD

**Issue by andres-erbsen** [coq/coq#10182]
#### Description of the problem

```coq
Require Import Coq.Arith.PeanoNat.
Goal forall A m pf1 pf2 (H : (fun _ : False => A) = (fun _ : False => A)),
match Nat.nlt_0_r m pf1 with end = match Nat.nlt_0_r m pf2 with end :> A.
Proof. intros. rewrite H.
(* Error: Anomaly "Not enough lambda/let's." *)
(* Please report at http://coq.inria.fr/bugs/. *)
```

Minimized from LambdaTamer 2009 export, which presumably compiled at the time... feel free to remove kind:regression if that seems like too much. On the other hand, this report may just be a duplicate of #3166 (from 2013) anyway...

#### Coq Version

8.9.0 and some 8.10+alpha compiled on Apr 8 2019

**Comment by herbelin** [comment]
The `rewrite` breaks the canonical form of the return predicate. The `rewrite` is not expected to be allowed there. Will be fixed by #9710 (compact representation of pattern-matching).

**Comment by ejgallego** [comment]
Was fixed likely by #13563 will add a test case.
